sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -196,6 +254,40 @@ case class RLike(left: Expression, right: Expression)
   override def escape(v: String): String = v
   override def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).find(0)
   override def toString: String = s"$left RLIKE $right"
+
+  override protected def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val patternClass = classOf[Pattern].getName
+
+    val literalRight: String = right match {
+      case x @ Literal(value: String, StringType) => escape(value)
+      case _ => null
+    }
```

Comment:
Okay got it. But this caches the value only if it's a literal. I think if we save the value in `mutableState` we could even use this "cache" if an expression returns the same result like the previous one.